pkg/registry/certificates/certificates/strategy.go

```diff
@@ -190,3 +194,48 @@ func (csrApprovalStrategy) PrepareForUpdate(ctx context.Context, obj, old runtim
 func (csrApprovalStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
 	return validation.ValidateCertificateSigningRequestUpdate(obj.(*certificates.CertificateSigningRequest), old.(*certificates.CertificateSigningRequest))
 }
+
+// GetAttrs returns labels and fields of a given object for filtering purposes.
+func GetAttrs(obj runtime.Object) (labels.Set, fields.Set, error) {
+	csr, ok := obj.(*certificates.CertificateSigningRequest)
+	if !ok {
+		return nil, nil, fmt.Errorf("not a certificatesigningrequest")
+	}
+	return labels.Set(csr.Labels), SelectableFields(csr), nil
+}
+
+// MatchCSR returns a generic matcher for a given label and field selector.
+func MatchCSR(label labels.Selector, field fields.Selector) storage.SelectionPredicate {
+	return storage.SelectionPredicate{
+		Label:       label,
+		Field:       field,
+		GetAttrs:    GetAttrs,
+		IndexFields: []string{"spec.signerName"},
```

Comment:
This has also been cargo-culted from the `pod` package too: https://github.com/kubernetes/kubernetes/blob/dd5272b76f07bea60628af0bb793f3cca385bf5e/pkg/registry/core/pod/strategy.go#L204

Again, I'm uncertain if it is required (or whether setting the PredicateFunc is even required) given that Pods support field selectors on many different fields, yet only lists `spec.nodeName` in the IndexFields list (linked above).

Can anyone shed any light as to whether this is required, or what the effect of setting it is?